---
title: Science
subtitle: Learning Resources
order: 5
author: A H
description: General Science, Biology, and Informatics 
image: '../images/home-page-3.jpeg'

---
---
A collection of self-guided learning resources created by others.
  
### Single-Cell Analyses

<details><summary>__Single-Cell RNA-seq__</summary>

[scRNA-seq Analysis Tutorial: Analysis of Single-Cell RNA-seq Data](https://github.com/cellgeni/scRNA.seq.course) -- A continuously updated and comprehensive single-cell RNA-seq analysis tutorial/course, __taught primarily in `R`__. This course begins with a discussion of single-cell methods, experimental design, and data processing and ends with single-cell dataset integration. It uses primarily `Seurat`, but also covers other tools for analysis and integration.  

[scRNA-seq Tutorial: Single-cell best practices](https://www.sc-best-practices.org/preamble.html) -- A very good tutorial on the best practices in single-cell RNA-seq analysis, __taught primarily in `Python`__. This course starts with pre-processing and QC, and ends with a brief overview on CITE-seq, immune repertoire, and integration. The [paper](https://www.embopress.org/doi/full/10.15252/msb.20188746) that this course was originally based on.

[Single-Cell Genomics Workshop: Single-Cell Genomics Day](https://satijalab.org/scgd22/) -- A yearly workshop by the Satija lab (leaders in the world of single-cell analysis) on various single-cell genomics analysis topics/methods. In addition to a "recent and future advances" session, the workshop covers spatial and temporal analysis, epigenomic analysis, genotype-phenotype landscapes, and multimodal analyses.

[scRNA-seq Best Practices Pipeline: nf-core/scrnaseq](https://nf-co.re/scrnaseq) -- A best-practices pipline for processing 10x genomics single-cell data using `Nextflow`, a workflow management tool that provides improved computational metics and reproducibility.

[Curated database of single cell studies](https://www.nxn.se/single-cell-studies/) -- A manually curated database of over 1800 single-cell studies, dating back to 2002. The doi, number of cells, organism, tissue, and experimental method are included, as well as other useful information.

</details>

<details><summary>__Single-Cell ATAC-seq__</summary>

[scATAC-seq Lectues: Best practices for ATAC-seq](https://liulab-dfci.github.io/bioinfo-combio/scatac.html) -- A set of lectures from Ming Tang that cover the scATAC-seq experimental method, pre-processing and QC, and anlysis and integration.

[scATAC-seq Review Paper: Single-cell ATAC sequencing analysis: From data preprocessing to hypothesis generation](https://www.sciencedirect.com/science/article/pii/S2001037020303019#f0005) -- A helpful review of scATAC-seq technologies and analysis software.

</details>

### Proteomics

[Mass-Spec Based Proteomics Aanalysis Tutorial]

[Best practices for mass-spec-based biomarker discovery](https://www.nature.com/articles/s41596-021-00566-6) -- A high-level paper overview of best practices for mass-spectrometry-based biomarker discovery. 

### Imaging Analysis
[Segmentation]


### General Bioinformatics

[Bioinformatics Course: Bioinformatics for Plant and Animal Sciences](https://www.youtube.com/playlist?list=PLhR2Go-lh6X5A5WbiO3SPHuoWbwpNznUl) -- A quality YouTube based course by Dr. Danny Arends (with [videos](https://www.youtube.com/playlist?list=PLhR2Go-lh6X5A5WbiO3SPHuoWbwpNznUl), access to [lectures](https://dannyarends.nl/bioinformatik/), [assignments](https://dannyarends.nl/bioinfo/), and answers) on bioinformatics. This course covers  biological topics (genetics, molecular biology, metabolism, homology/phylogeny, etc.), as well as computational and analysis topics (transcriptomics, R, stats) and ends with learning how to create an R package.

[Rosalind](https://rosalind.info/problems/locations/) -- a great resource to learn about bioinformatics (including algorithms) and programming by problem-solving.


### Statistics

[Introductory Stats Course: Introduction to Experimental Design and Hypothesis Testing](https://github.com/gladstone-institutes/Bioinformatics-Workshops/wiki/Introduction-to-Experimental-Design-and-Hypothesis-testing) -- A beginner-friendly set of resources (slides, code) from an online workshop by the Gladstone Institutes to work though. The resources cover concepts underlying hypothesis testing.

[Intermediate Course: Statistics of Enrichment Analyses Methods](https://github.com/gladstone-institutes/Bioinformatics-Workshops/wiki/Statistics-of-Enrichment-Analyses-Methods) -- A course designed for those familiar with basic statistics and experimental design concepts, an understanding of high-throughput analyses (RNA-seq, Mass Spec, etc.), and a working knowledge of R.

