---
title: "Installation and Resources"
order: 12
---

### Software

**Bash Shell**
For this series, you will need access to the Bash shell. `Linux` and `macOS` users will have access to this already and should not need to install anything. 

`Windows` users will need to do some extra work. 

- Install `Git for Windows`, which will provide acess to `Bash shell` and `Git` by following the instructions [here](https://carpentries.github.io/workshop-template/#shell).

**RNA-seq processing workflow**

In the final lesson, we will use `miniconda` to control in installation of several tools. Follow the steps below.

1. Install miniconda using the appropriate directions from here:
<https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html>

2. After closing and re-opening the terminal, the `conda` base environment should be enabled. To check this, make sure you see `(base)` to the left of your user name in the terminal prompt.

	```
	(base) user: $
	```

3. Create a `conda` environment called `rnaseq` while installing the needed tools/software -- enter "y" when prompted.

	```
	conda activate base
	conda create -n rnaseq -c bioconda fastqc trim-galore subread star
	```

### Data

**Introductory Files**
These data are zipped and should begin downloading once the link below is clicked.

- Click [here](https://swcarpentry.github.io/shell-novice/data/shell-lesson-data.zip) to download the data.
- Move the `shell-lesson-data.zip` file from your Downloads to your Desktop.
- Unzip the file to extract the data. You should end up with a `shell-lesson-data` directory on your Desktop.

**RNA-seq similated data/files**
For the last lesson, you will need simulated reads, fasta files, index files, etc.

- Click [here](https://drive.google.com/drive/folders/1w7CW7UCZPKg296MIpIAEu08DgFvhjsGJ?usp=share_link) to download the data.
- Move the directory to your Desktop if possible.