<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Self-Guided Learning - Learning Resources</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/cookie-consent/cookie-consent.js"></script>
<link href="site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"simple",
  "consent_type":"implied",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary","functionality","tracking","targeting"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  });
});
</script> 
  
<style>html{ scroll-behavior: smooth; }</style>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Self-Guided Learning</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./software-installs.html">
 <span class="menu-text">Software Installs</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./learning-resources.html" aria-current="page">
 <span class="menu-text">Learning Resources</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./domain-specific-series.html">
 <span class="menu-text">Domain-Specific Series</span></a>
  </li>  
</ul>
              <div class="quarto-toggle-container">
                  <a href="" class="quarto-reader-toggle nav-link" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
              </div>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#languages-and-programming" id="toc-languages-and-programming" class="nav-link active" data-scroll-target="#languages-and-programming">Languages and Programming</a></li>
  <li><a href="#plotting-and-visualizations" id="toc-plotting-and-visualizations" class="nav-link" data-scroll-target="#plotting-and-visualizations">Plotting and Visualizations</a></li>
  <li><a href="#single-cell-analyses" id="toc-single-cell-analyses" class="nav-link" data-scroll-target="#single-cell-analyses">Single-Cell Analyses</a></li>
  <li><a href="#proteomics" id="toc-proteomics" class="nav-link" data-scroll-target="#proteomics">Proteomics</a></li>
  <li><a href="#imaging-and-microscopy" id="toc-imaging-and-microscopy" class="nav-link" data-scroll-target="#imaging-and-microscopy">Imaging and Microscopy</a></li>
  <li><a href="#open-science-reproducibility" id="toc-open-science-reproducibility" class="nav-link" data-scroll-target="#open-science-reproducibility">Open Science &amp; Reproducibility</a></li>
  <li><a href="#artificial-intelligence-machine-learning" id="toc-artificial-intelligence-machine-learning" class="nav-link" data-scroll-target="#artificial-intelligence-machine-learning">Artificial Intelligence &amp; Machine Learning</a></li>
  <li><a href="#general-bioinformatics" id="toc-general-bioinformatics" class="nav-link" data-scroll-target="#general-bioinformatics">General Bioinformatics</a></li>
  <li><a href="#statistics" id="toc-statistics" class="nav-link" data-scroll-target="#statistics">Statistics</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-page-left" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Learning Resources</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>A collection of self-guided learning resources created by others.</p>
<section id="languages-and-programming" class="level3">
<h3 class="anchored" data-anchor-id="languages-and-programming">Languages and Programming</h3>
<details>
<summary>
<strong>Python</strong>
</summary>
<p><a href="https://www.youtube.com/watch?v=mvK0UzFNw1Q">How to Learn to Code</a> – A 5 minute video on how to approach programming, either learning theory, soliciting advice, OR (hint hint) <strong><em>problem solving</em></strong>.</p>
<p><a href="https://online.umich.edu/courses/programming-for-everybody-getting-started-with-python/">Python Course: Programming for Everybody (Getting Started with Python)</a> – A terrific free course by the University of Michigan to learn the basics of programming in <code>Python</code>, with no prerequisites.</p>
<p><a href="https://www.youtube.com/watch?v=ocA2IMe7dpA&amp;list=PLhYDP66xNTKRdtUM8ekEmq1fC0mosXrgn&amp;index=15">Biopython Course: Bioinformatics with Biopython</a> – A 1-hour YouTube course that covers some common uses of biopython. <code>Biopython</code> is a set of tools for biological computation that is written in <code>Python</code>. It has many useful functions for processing and working with sequence files.</p>
<p><a href="https://www.pythoncheatsheet.org/">Python Cheatsheet</a> – A quick-reference, cheatsheet-styled website for common tasks/tricks in <code>Python</code>. Helpful for beginners and power-users.</p>
</details>
<details>
<summary>
<strong>R</strong>
</summary>
<p><a href="https://swirlstats.com/">R Course: swirl - learn R, in R</a> – Super cool way to learn <code>R</code> programming and datascience interactively, at your own pace, and in the <code>R</code> console. There is a library of courses you can learn from.</p>
<p><a href="https://www.youtube.com/playlist?list=PLhR2Go-lh6X4fCAa3c_TBAjZD5A25s7fo">R Course: Data Analysis using R</a> – A comprehensive and high quality YouTube based course by Dr.&nbsp;Danny Arends (with videos, access to lectures, assignments, and answers) for those within minimal to no previous programming experience. This might be one of the best courses available.</p>
<p><a href="https://r4ds.had.co.nz/index.html">R for Data Science</a> – A website for the “R for Data Science” book, with a focus on how to perform data science with <code>R</code>, from data exploration (visualization, workflow, working with tidy data, scripting), wrangling, modeling, and communicating results (<code>R markdown</code>).</p>
<p><a href="https://cran.r-project.org/doc/contrib/Baggott-refcard-v2.pdf">R Cheatsheet</a> – A quick-reference cheatsheet for <code>R</code>.</p>
</details>
<details>
<summary>
<strong>Unix and Bash</strong>
</summary>
<p><a href="https://seankross.com/the-unix-workbench/introduction.html">Unix Book: Unix Workbench</a> – A website (book) for those new to Unix-like operating systems and working at the command-line. This book covers unix and command-line basics, as well as introductory bash programming concepts (math, variables, loops, input/output, arrays, pbraces, functions), and writing programs. As a nice bonus, it also gives brief introductions to <code>Git</code>, <code>GitHub</code>, and Cloud Computing.</p>
<p><a href="https://github.com/Idnan/bash-guide">Bash Reference: bash-guide</a> – Common commands, with some minimal examples.</p>
</details>
<details>
<summary>
<strong>Julia</strong>
</summary>
<p><a href="https://github.com/Datseris/Zero2Hero-JuliaWorkshop">Julia Course: Zero-to-Hero Julia Workshop</a> – Two day intensive, recorded (<a href="https://www.youtube.com/watch?v=Fi7Pf2NveH0">YouTube</a>), workshop on the programming language, <code>Julia</code>. This workshop is intended for beginners. <code>Julia</code> is a programming language designed for scientific computing, to be quick, but also readable and high-level like <code>Python</code>.</p>
</details>
</section>
<section id="plotting-and-visualizations" class="level3">
<h3 class="anchored" data-anchor-id="plotting-and-visualizations">Plotting and Visualizations</h3>
<details>
<summary>
<strong>With Python</strong>
</summary>
<p><a href="https://gilberttanner.com/blog/introduction-to-data-visualization-inpython/">Introduction to Data Visualization in Python</a> – A very simple introduction to data visualization in <code>Python</code>.</p>
<p><a href="https://realpython.com/tutorials/data-viz/">Python Data Visualization Tutorials</a> – A set of tutorials for plotting with <code>Python</code> in <code>Pandas</code>, <code>matplotlib</code>, <code>seaborn</code>, and <code>bokeh</code> (interactive visualizations).</p>
<p><a href="https://www.kaggle.com/code/kanncaa1/plotly-tutorial-for-beginners">Interactive Visualizations with Plotly in Python</a> – An introduction, for beginners, to interactively plotting in <code>Python</code> using <code>Plotly</code>.</p>
</details>
<details>
<summary>
<strong>With R</strong>
</summary>
<p><a href="https://rkabacoff.github.io/datavis/">Data Visualization with R</a> – A Web-based book focused on data visualization and plotting with <code>R</code>. It covers data importing, cleaning, using plotting libraries (<code>ggplot2</code>), customizing plots, interactive visualization, and more.</p>
<p><a href="https://shiny.rstudio.com/images/shiny-cheatsheet.pdf">Shiny Cheatsheet</a> – A quick reference cheatsheet for creating interactive web apps with <code>shiny</code>.</p>
</details>
</section>
<section id="single-cell-analyses" class="level3">
<h3 class="anchored" data-anchor-id="single-cell-analyses">Single-Cell Analyses</h3>
<details>
<summary>
<strong>Single-Cell RNA-seq</strong>
</summary>
<p><a href="https://github.com/cellgeni/scRNA.seq.course">scRNA-seq Analysis Tutorial: Analysis of Single-Cell RNA-seq Data</a> – A continuously updated and comprehensive single-cell RNA-seq analysis tutorial/course, <strong>taught primarily in <code>R</code></strong>. This course begins with a discussion of single-cell methods, experimental design, and data processing and ends with single-cell dataset integration. It uses primarily <code>Seurat</code>, but also covers other tools for analysis and integration.</p>
<p><a href="https://www.sc-best-practices.org/preamble.html">scRNA-seq Tutorial: Single-cell best practices</a> – A very good tutorial on the best practices in single-cell RNA-seq analysis, <strong>taught primarily in <code>Python</code></strong>. This course starts with pre-processing and QC, and ends with a brief overview on CITE-seq, immune repertoire, and integration. The <a href="https://www.embopress.org/doi/full/10.15252/msb.20188746">paper</a> that this course was originally based on.</p>
<p><a href="https://satijalab.org/scgd22/">Single-Cell Genomics Workshop: Single-Cell Genomics Day</a> – A yearly workshop by the Satija lab (leaders in the world of single-cell analysis) on various single-cell genomics analysis topics/methods. In addition to a “recent and future advances” session, the workshop covers spatial and temporal analysis, epigenomic analysis, genotype-phenotype landscapes, and multimodal analyses.</p>
<p><a href="https://nf-co.re/scrnaseq">scRNA-seq Best Practices Pipeline: nf-core/scrnaseq</a> – A best-practices pipline for processing 10x genomics single-cell data using <code>Nextflow</code>, a workflow management tool that provides improved computational metics and reproducibility.</p>
<p><a href="https://www.nxn.se/single-cell-studies/">Curated database of single cell studies</a> – A manually curated database of over 1800 single-cell studies, dating back to 2002. The doi, number of cells, organism, tissue, and experimental method are included, as well as other useful information.</p>
</details>
<details>
<summary>
<strong>Single-Cell ATAC-seq</strong>
</summary>
<p><a href="https://liulab-dfci.github.io/bioinfo-combio/scatac.html">scATAC-seq Lectues: Best practices for ATAC-seq</a> – A set of lectures from Ming Tang that cover the scATAC-seq experimental method, pre-processing and QC, and anlysis and integration.</p>
<p><a href="https://www.sciencedirect.com/science/article/pii/S2001037020303019#f0005">scATAC-seq Review Paper: Single-cell ATAC sequencing analysis: From data preprocessing to hypothesis generation</a> – A helpful review of scATAC-seq technologies and analysis software.</p>
</details>
</section>
<section id="proteomics" class="level3">
<h3 class="anchored" data-anchor-id="proteomics">Proteomics</h3>
<details>
<summary>
<strong>With R</strong>
</summary>
<p><a href="https://rformassspectrometry.github.io/docs/">R for Mass Spectrometry</a> – A very nice and well documented resource demonstrating how to use flexible and relablle <code>R</code> software for the analysis and interpretation of high throughput mass spectrometry assays, including proteomics and metabolomics experiments.</p>
<p><a href="https://pnnl-comp-mass-spec.github.io/proteomics-data-analysis-tutorial/">Proteomics Tutorial in R</a> – A website-based tutorial covering from initial processing to differential and pathway analyses.</p>
</details>
<details>
<summary>
<strong>Best Practices</strong>
</summary>
<p><a href="https://portlandpress.com/biochemist/article/42/5/64/226371/A-beginner-s-guide-to-mass-spectrometry-based">Beginner’s Guide to Mass Spectrometry</a> – A beginner-oriented introduction to sample preparation, mass spectrometry and data analysis.</p>
<p><a href="https://www.nature.com/articles/s41596-021-00566-6">Best practices for mass-spectrometry-based biomarker discovery</a> – A high-level paper overview of best practices for mass-spectrometry-based biomarker discovery.</p>
</details>
<details>
<summary>
<strong>In Galaxy</strong>
</summary>
<p><a href="https://training.galaxyproject.org/training-material/topics/proteomics/">Proteomics in Galaxy</a> – A set of training sources focused on proteomics workflows in Galaxy.</p>
</details>
</section>
<section id="imaging-and-microscopy" class="level3">
<h3 class="anchored" data-anchor-id="imaging-and-microscopy">Imaging and Microscopy</h3>
<details>
<summary>
<strong>Introduction to Bioimage Analysis</strong>
</summary>
<p><a href="https://www.edx.org/course/image-processing-and-analysis-for-life-scientists">MOOC: Image Processing and Analysis for Life Scientists</a> – A online 7-week, self-paced course covering state of the art image analysis strategies.</p>
<p><a href="https://bioimagebook.github.io/index.html">Book Course: Introduction to Bioimage Analysis</a> – A very well structured book designed to bring biologists up to speed on image analysis while serving as a practical guide to bioimage analysis with <code>Fiji</code> and <code>Python</code>.</p>
<p><a href="https://www.ibiology.org/techniques/bioimage-analysis/">Video Series: Introduction to bioimage analysis</a> – A nice video introduction to bioimage analysis, covering pre-processing, segmentation, tracking, measurement and pheontype classification, and tips and best practices. A great place to start.</p>
<p><a href="https://febs.onlinelibrary.wiley.com/doi/full/10.1002/1873-3468.14451">Paper: A Hitchhiker’s guide through the bio-image analysis software universe</a> – Guidance on how to choose the bioimage analysis software, and what to consider when making that decisions (skills, data type, time, budget, etc.).</p>
</details>
<details>
<summary>
<strong>With ImageJ &amp; Fiji</strong>
</summary>
<p><a href="https://www.youtube.com/playlist?list=PL5ESQNfM5lc7SAMstEu082ivW4BDMvd0U">Bioimage analysis course in Fiji</a> – A YouTube course by Robert Haase in bioimage analysis using <code>Fiji</code>, a very commonly used open-source image analysis tool.</p>
<p><a href="https://www.youtube.com/playlist?list=PL5Edc1v41fyCLFZbBCLo41zFO-_cXBfAb">Fiji Tutorials</a> – An YouTube series covering the basics of working with <code>Fiji</code>, from importing data, to counting cells, to working with movies and batch converting files.</p>
</details>
<details>
<summary>
<strong>With Python</strong>
</summary>
<p><a href="https://guiwitz.github.io/PyImageCourse_beginner/README.html">Beginner Course: Image processing with Python for Beginners</a> – An online, hands-on, web-book-based, course that covers basic image processing using <code>Python</code> and the scientific packages <code>Numpy</code>, <code>scikit-image</code>, <code>Matplotlib</code>, and <code>Pandas</code>.</p>
<p><a href="https://github.com/guiwitz/PyImageCourse">Intermediate Course: Image processing with Python for Intermediate Users</a> – An online <code>Juypter Notebook</code>-based course covering intermediate to advanced topics in image processing using <code>Python</code>.</p>
<p><a href="https://www.youtube.com/playlist?list=PLHae9ggVvqPgyRQQOtENr6hK0m1UquGaG">Video Series: Introductory python tutorials for image processing</a> – A series of YouTube videos designed for absolute beginners on Python coding, with an emphasis on image analysis.</p>
<p><a href="https://www.youtube.com/playlist?list=PLZsOBAyNTZwanH91tjVOPJ7zpUiLFqM-F">Denoising Images using Python</a> – A YouTube series on denoising images using python, with a focus on microscopy and scientific imaging. Traditional approaches (Gaussaian, median, bilateral, total variation, non-local means, and BM3D) as well as deep learning techniques (autoencoders and <code>Noise2Void</code>) are covered.</p>
</details>
<details>
<summary>
<strong>With napari</strong>
</summary>
<p><a href="https://github.com/sofroniewn/napari-training-course">napari-training-course</a> – A <code>Jupyter Notebook</code>-based <code>napari</code> training course covering how to visualize bioimages, perform manual annotations, and do interactive analyses.</p>
<p><a href="http://alisterburt.com/napari-workshops/home.html">napari-workshops</a> – A set of <code>napari</code> workshops, with hands-on examples to work through, covering bioimage visualization, manual annotation, nuclei segmentation (<code>cellpose napari plugin</code> and <code>stardist napari plugin</code>), using custom colormaps, interactive analysis (<code>Jupyter Notebook</code>, <code>napari</code>, <code>scikit-image</code>, and <code>scipy</code>), creating a <code>napari</code> plugin,</p>
<p><a href="https://napari.org/stable/tutorials/index.html">napari-tutorials</a> – A set of <code>napari</code> tutorials for those interested in using napari. The tutorials cover how to do basic annotation, image processing, segmentation, and tracking</p>
<p><a href="https://napari.org/stable/howtos/index.html">napari-how-to guides</a> – <code>Napari</code> guides for current users of <code>napari</code>, explaining how ot use layer, hook up events, work with <code>ImageJ</code>, run <code>napari</code> in <code>Docker</code>, monitor performance, and run <code>napari</code> headlessly.</p>
</details>
<details>
<summary>
<strong>Microscopy</strong>
</summary>
<p><a href="https://www.ibiology.org/online-biology-courses/microscopy-series/">Microscopy Series</a> – A video series by iBiology on all aspects of microscopy.</p>
<p><a href="https://myscope.training/">Interactive training: Methods in Microscopy</a> – A cool interactive training course on many microscopy techniques.</p>
<p><a href="https://www.nature.com/articles/s41596-020-0313-9">Paper: Tutorial - guidance for quantitative confocal microscopy</a> – A paper tutorial covering confocal microscopy image acquistition (sample preparation, microscope choice, capture configuration and parameters), common pitfalls (photobleaching, cross-talk, etc.) and how to avoid them, and guidelines on analysis and presentation.</p>
<p><a href="http://cryo-em-course.caltech.edu/cryoem_welcome">Interactive training: Course on cryo-EM and 3D-EM</a> – An interactive course from CalTech that covers fundamental concepts, sample preparation, and data processing in cryo-EM and 3D-EM.</p>
</details>
</section>
<section id="open-science-reproducibility" class="level3">
<h3 class="anchored" data-anchor-id="open-science-reproducibility">Open Science &amp; Reproducibility</h3>
<details>
<summary>
<strong>Open Science</strong>
</summary>
<p><a href="https://posit.co/resources/cheatsheets/?type=posit-cheatsheets/">Posit Cheatsheets</a> – An amazing list of cheatsheets for eveyrthing <code>rmarkdown</code>, <code>shiny</code>, <code>reticulate</code>, <code>stringr</code>, <code>readr</code>, <code>sparkly</code>, <code>tidyr</code>, <code>dplyr</code>, <code>ggplot2</code>, and more.</p>
</details>
<details>
<summary>
<strong>Reproducibility</strong>
</summary>
</details>
</section>
<section id="artificial-intelligence-machine-learning" class="level3">
<h3 class="anchored" data-anchor-id="artificial-intelligence-machine-learning">Artificial Intelligence &amp; Machine Learning</h3>
<details>
<summary>
<strong>Introductory</strong>
</summary>
<p><a href="https://developers.google.com/machine-learning/crash-course">Hands-On Crash Course: Introduction to Machine Learning</a> – A foundational course from Google that with exercises, lessons, and interactive visualizations. In this course you will learn and apply machine learning concepts.</p>
<p><a href="https://developers.google.com/machine-learning/intro-to-ml">Introduction to Machine Learning</a> – From Google, a very short introduction to machine learning concepts (does not cover how to implement or work with data).</p>
<p><a href="https://developers.google.com/machine-learning/problem-framing">Introduction to Machine Learning Problem Framing</a> – From Google, learn how to determine when machine learning is a proper approach for a ploblen and how to outline a machine learning solution.</p>
<p><a href="https://developers.google.com/machine-learning/data-prep">Data Preparation and Feature Engineering</a> – From Google, learn how to construct and transform your data for use in machine learning workflows.</p>
<p><a href="https://developers.google.com/machine-learning/testing-debugging">Testing and Debugging</a> – From Gogle, learn how to train a model and use that model to make predictions. The course incluses topics from debugging your model to monitoring your pipeline in production.</p>
<p><a href="https://www.edx.org/course/artificial-intelligence-ai">kaggle: Artificial Intelligence</a> – A MOOC on edx (ColumbiaX) that teaches the fundamental of artificial intelligence and how to apply them.</p>
<p><a href="https://www.kaggle.com/learn/intro-to-machine-learning">kaggle: Intro to Machine Learning</a> – A course by kaggle that teaches the core ideas in machine learning and helps you build your first models.</p>
</details>
<details>
<summary>
<strong>Advanced</strong>
</summary>
<p><a href="https://www.kaggle.com/learn/intermediate-machine-learning">kaggle: Intermediate Machine Learning</a> – A course by kaggle that focuses on handling missing values, non-numeric values, data leakage, and more.</p>
<p><a href="https://www.kaggle.com/learn/computer-vision">kaggle: Computer Vision</a> – A course by kaggle that teaches you the vasics of computer vision and how to build convolutional neural networks. You will be using <code>TensorFlow</code> and <code>Keras</code>.</p>
</details>
</section>
<section id="general-bioinformatics" class="level3">
<h3 class="anchored" data-anchor-id="general-bioinformatics">General Bioinformatics</h3>
<details>
<summary>
<strong>Introductory</strong>
</summary>
<p><a href="https://www.youtube.com/playlist?list=PLhR2Go-lh6X5A5WbiO3SPHuoWbwpNznUl">Bioinformatics Course: Bioinformatics for Plant and Animal Sciences</a> – A quality YouTube based course by Dr.&nbsp;Danny Arends (with <a href="https://www.youtube.com/playlist?list=PLhR2Go-lh6X5A5WbiO3SPHuoWbwpNznUl">videos</a>, access to <a href="https://dannyarends.nl/bioinformatik/">lectures</a>, <a href="https://dannyarends.nl/bioinfo/">assignments</a>, and answers) on bioinformatics. This course covers biological topics (genetics, molecular biology, metabolism, homology/phylogeny, etc.), as well as computational and analysis topics (transcriptomics, <code>R</code>, stats) and ends with learning how to create an <code>R</code> package.</p>
<p><a href="https://rosalind.info/problems/locations/">Rosalind</a> – A great resource to learn about bioinformatics (including algorithms) and programming by problem-solving.</p>
</details>
</section>
<section id="statistics" class="level3">
<h3 class="anchored" data-anchor-id="statistics">Statistics</h3>
<details>
<summary>
<strong>Introductory</strong>
</summary>
<p><a href="https://github.com/gladstone-institutes/Bioinformatics-Workshops/wiki/Introduction-to-Experimental-Design-and-Hypothesis-testing">Introductory Stats Course: Introduction to Experimental Design and Hypothesis Testing</a> – A beginner-friendly set of resources (slides, code) from an online workshop by the Gladstone Institutes to work though. The resources cover concepts underlying hypothesis testing.</p>
</details>
<details>
<summary>
<strong>Advanced</strong>
</summary>
<p><a href="https://github.com/gladstone-institutes/Bioinformatics-Workshops/wiki/Statistics-of-Enrichment-Analyses-Methods">Intermediate Course: Statistics of Enrichment Analyses Methods</a> – A course designed for those familiar with basic statistics and experimental design concepts, an understanding of high-throughput analyses (RNA-seq, Mass Spec, etc.), and a working knowledge of <code>R</code>.</p>
</details>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
      <div class="nav-footer-center"><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
  </div>
</footer>



<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>